### PR TITLE
[xcvrd] Ignore processing ports that are not front-panel

### PR DESCRIPTION
**Pull request by vivekrnv**
#### Description

Ignore processing the front panel ports in xcvrd. This include ports with suffixes other than Ethernet<num> and ports that have role field set to any value other than Ext. 

Requires: https://github.com/sonic-net/sonic-buildimage/pull/18465

### CODE REVIEW

**Review by dgsudharsan — APPROVED** [review]

File: sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py [hunk 1]
```diff
@@ -74,6 +74,17 @@ def apply_filter_to_fvp(self, filter, fvp):
                 if key not in (set(filter) | set({'index', 'port_name', 'asic_id', 'op'})):
                     del fvp[key]
 
+    def refresh_role_map(self):
+        for ns in self.namespaces:
+            cfg_db = daemon_base.db_connect("CONFIG_DB", namespace=ns)
+            port_table = swsscommon.Table(cfg_db, swsscommon.CFG_PORT_TABLE_NAME)
+            for key in port_table.getKeys():
+                print("key, {}".format(key))
```
Comment (dgsudharsan) [inline sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py:82]: Can you remove the print

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py (hunk 1 above)
Comment (vivekrnv) [reply sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py]: updated

**Review by dgsudharsan — APPROVED** [review]

**Review by mihirpat1 — COMMENTED** [review]
@vivekrnv Can you please help in addressing the built failure?

File: sonic-xcvrd/tests/test_xcvrd.py [hunk 2]
```diff
@@ -826,6 +826,60 @@ def test_is_error_sfp_status(self):
         assert not is_error_block_eeprom_reading(int(SFP_STATUS_INSERTED))
         assert not is_error_block_eeprom_reading(int(SFP_STATUS_REMOVED))
 
+
+    @patch('swsscommon.swsscommon.Select.addSelectable', MagicMock())
+    @patch('swsscommon.swsscommon.Table')
+    @patch('swsscommon.swsscommon.SubscriberStateTable')
+    @patch('swsscommon.swsscommon.Select.select')
+    def test_handle_pude_front_panel(self, mock_select, mock_sub_table, mock_swsscommon_table):
```
[the remark this answers is not in the document]
Comment (vivekrnv) [reply sonic-xcvrd/tests/test_xcvrd.py:834]: Good catch, yes will update

File: sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py [hunk 3]
```diff
@@ -119,9 +129,19 @@ def handle_port_update_event(self):
                     (port_name, op, fvp) = port_tbl.pop()
                     if not port_name:
                         break
-                    if not validate_port(port_name):
-                        continue
+ 
                     fvp = dict(fvp) if fvp is not None else {}
+                    role = fvp.get(multi_asic.PORT_ROLE, None)
```
Comment (prgeor) [inline sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py:134]: @vivekrnv Why not always get the `role` from self.port_role_map{} which is updated in the __init__ with the role form config DB?

File: sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py [hunk 4]
```diff
@@ -119,9 +129,19 @@ def handle_port_update_event(self):
...
+                    if role:
+                        self.port_role_map[port_name] = role
+                    else:
+                        # role attribute might not be present for state DB
```
Comment (prgeor) [inline sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py:138]: @vivekrnv at line 85 the map is updated from config_db during __init__

File: sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py (hunk 3 above)
Comment (vivekrnv) [reply sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py:134]: This is to handle DPB use case

File: sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py (hunk 4 above)
Comment (vivekrnv) [reply sonic-xcvrd/xcvrd/xcvrd_utilities/port_event_helper.py:138]: Yes, but to handle dpb we need to also check for presence of role attribute here

**Review by mihirpat1 — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@prgeor can we go a head and merge?

**Review by prgeor — APPROVED** [review]

**Comment by prgeor** [comment]
> @prgeor can we go a head and merge?

Yes

**Pull request merged**